### DISCUSSION TITLE
Questions to general BIRT Framework migration strategy

### DISCUSSION THREAD

**Discussion by michaelabsb** [eclipse-birt/birt#1854] [General]
Hello BIRT community, 

we are currently migrating our web application to a newer JDK version (17 or 21 – there is no final decision jet). At the same time, we are trying to update our tools. Among others we are evaluating a new BIRT version.

**Our current situation:**

For production we use BIRT 4.5 runtime library. For report development we use an old eclipse Mars version running on JDK 1.8 and BIRT 4.5 Framework.

**The Problem:**

The current BIRT version 4.16 doesn’t seem to be a stable version. The troubles start when trying to use the current BIRT designer. Even the precompiled BIRT-Designer-All-In-One version fails to open reports containing Cross Tabs. Creating new reports and adding Corss Tab fails immediately. The report viewer in this version fails also.

**Our questions:**

1.	Do you recommend to stick to the running BIRT 4.5 version? Is this version compatible to JDK 17 and/or to JDK 21? Is there any experience in using BIRT 4.5 in JDK 17 or JDK 21 environment?
2.	What is the future of BIRT? Will there be a stable, production grade version again?
3.	Is it more advisable to move to other report generation framework?

Thank you for your opinion!
Michael

**Comment by merks** [comment]
You can't stay in the past forever, though it appears you've done that for a long time so far.   

The 4.5 version is not supported so whether that works in Java 11, 17 or 21 is up to you to determine and decide.

If there are problems, you should report them with enough information to reproduce them so they will get fixed in the 4.17 release rather than assert that no release is of production grade (which is kind of implicitly insulting, though I'm sure you didn't intend it that way).

It's best when reporting a problem to test a nightly build whether the problem is reproducible there before it's released:

https://download.eclipse.org/birt/updates/nightly/latest/index.html

Your question 3 is like asking Apple if it's advisable switch to Microsoft?  😜 

I wonder where else you can get so much free goodness.  Also, I wonder how much work it would be to switch from what you have to that other free goodness?

**Reply by michaelabsb** [reply]
Hi Ed, thanks a lot for your quick response :-)

**Reply by speckyspooky** [reply]
Currently BIRT is one of the generic frameworks to create different output formats based on the same template.
All other tools like PowerBI and all the big tools and names are not open source and licence costs are required of it.
From technical side BIRT with the Java environment is nevertheless of his age one of the best tools for integratoin.
And the script combinateion of JavaScrpt & Java is only given here.
Yes also some Python-libraries are powerfull but all in all with the integration and standardization BIT is one of the best.

Therefore I can only agree to @merks it is a very good decision to go forward.
I swithced my production environment from 4.6 to 4.14 this April and all is fine.
We plan this year to migrate in steps from 4.14 to 4.15 and 4.16 or may be 4.17 - dependencies of other components are our limitations here.

And yes, I use also cross tables and my reports work as well from 4.6 at 4.14.
But cross tables are very hard for support and development.

And doubled agree to Ed, if you have trouble with reports than please don't hesitate to open issues
in best case with a small example without external librares and dependencies so that we can debug your problems.

From my side I can say that the version 4.16 is a stable version(!)
But the jump from 4.5 to 4.16 is more like a small jump.

But you will get them and I'm optimistic that the issues/topics can be solved through your team and the community.

**Reply by speckyspooky** [reply]
@michaelabsb 
May be you got with BIRT 4.16 this issue **"Creating/Editing datacubes leads to error #1824"** which will be fixed with BIRT 4.17

**Reply by michaelabsb** [reply]
Thank you Thomas, for your comment and the hint. The nightly build of BIRT 4.17 fixes the Corss Tab problem. In this case, we look forward to the release of BIRT 4.17.

**Comment by michaelabsb** [comment]
By the way, is possible to get the fix for the Corss Tab problem (#1824) in the 4.16 Version?

**Reply by merks** [reply]
What’s possible and what will actually happen are two different things.  It’s possible if you do it yourself.  The 4.17 release will actually happen around the second week of September.
